internal/services/loganalytics/log_analytics_data_export_resource.go

```diff
@@ -184,3 +205,37 @@ func flattenDataExportDestination(input *operationalinsights.Destination) string
 
 	return resourceID
 }
+
+func expandDataExportDestinationMetaData(input []interface{}) *operationalinsights.DestinationMetaData {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+
+	metaData := input[0].(map[string]interface{})
+
+	return &operationalinsights.DestinationMetaData{
+		EventHubName: utils.String(metaData["eventhub_name"].(string)),
+	}
+}
+
+func flattenDataExportDestinationMetaData(input *operationalinsights.Destination) []interface{} {
+	if input == nil || input.DestinationMetaData == nil {
+		return []interface{}{}
+	}
+
+	// As API always returns `"metaData": {}` when it isn't specified, so here it has to return `[]interface{}{}` otherwise TF would cause diff
+	if input.DestinationMetaData.EventHubName == nil {
+		return []interface{}{}
+	}
+
+	var eventHubName string
+	if input.DestinationMetaData.EventHubName != nil {
```

Comment:
the name is destination metadata? is this metadata added to the destination OR is it the final destination for the data?

Reply:
It's destination metadata. The whole block means the data from all tables in the rule are sent to a specific Event Hub when an Event Hub name is selected. See more details from https://docs.microsoft.com/en-us/azure/azure-monitor/logs/logs-data-export?tabs=portal#create-or-update-data-export-rule.